### ISSUE TITLE
`preferred-...` properties and Rust lib's `Window::set_size()` ineffective

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#6535]
Consider [this SlintPad demo](https://slintpad.com/?snippet=import+%7B+Button+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+Demo+%7B%0A++++preferred-width%3A+500px%3B%0A++++preferred-height%3A+500px%3B%0A%0A++++Button+%7B%0A++++++++%2F%2F+Commenting+this+property+out%2C+or+making+it+an+empty%0A++++++++%2F%2F+string+makes+the+component+tiny+and+unresizable.%0A++++++++text%3A+%22Foo%22%3B%0A++++%7D%0A%7D%0A&style=native) (see comment):

```slint
import { Button } from "std-widgets.slint";

export component Demo {
    preferred-width: 500px;
    preferred-height: 500px;

    Button {
        // Commenting this property out, or making it an empty
        // string makes the component tiny and unresizable.
        text: "Foo";
    }
}
```

I got to this code by reducing my app drastically. My actual use case is calling `Window::set_size()` on the Rust side before `app_window.show()`, which stays ineffective just like when using the `preferred-...` properties. In contrast to this example, though, my app window behaves normal (displaying in preferred, i.e., minimal size, and being normally resizable); `Window::set_size()` is simply ineffective, preventing me to restore the size saved on last app termination.

Above example on Windows 10 with `text` property being set to `"Foo"`:

https://github.com/user-attachments/assets/e3daa5c3-ab7f-4101-aa67-4e9e59c9a110

Same example with `text` property commented out:

https://github.com/user-attachments/assets/3b690b9d-6aa2-43b4-9fb4-7377dee253e0

**Comment by ogoffart** [comment]
indeed, there is a bug there. Looks like the compiler override the explicit preferred size with that of the content.

**Comment by dfaure-kdab** [comment]
Due to `export component Demo {`, this testcase (nowadays, at least) leads to the warning `warning: Exported component 'Demo' doesn't inherit Window. This is deprecated`.
When changing that line to `export component Demo inherits Window`, not only does the warning disappear, but the bug disappears too.

Claude says the reason is `When Demo doesn't inherit Window, the compiler wraps it, and Demo's root element gets width: parent.width from ExpandsToParentGeometry, which then suppresses the preferred-width`. I didn't verify this yet.
But I have to wonder, is it worth fixing a bug in a deprecated construct?

**Comment by ogoffart** [comment]
Even if it is deprecated, it is still what happens when previewing a single component.
i think the preferred size ought to be respected, even when an implicit Window is being created.

But it is not a high priority bug.

**Comment by tilladam** [comment]
Still reproduces on current master (`a84bb1dd52`, 1.18.0), and the numbers pin down exactly what @dfaure-kdab observed.

Both files are identical apart from the single word `inherits Window` on the component. Window size measured at DPR 2 and reported in logical pixels:

| snippet | resulting window | `preferred-width/height: 500px` |
| --- | --- | --- |
| `export component Demo {` | **45 x 32** | **ignored** |
| `export component Demo inherits Window {` | **500 x 500** | honoured |

45x32 is just the `Button`'s own natural size, so in the implicit-`Window` case the declared preferred size is not merely overridden — it contributes nothing at all. That matches @ogoffart's reading ("the compiler override the explicit preferred size with that of the content") and confirms @dfaure-kdab's finding that adding `inherits Window` makes it go away.

Which leaves this exactly where @ogoffart put it: even though the bare-component form is deprecated, it is still what a single-component preview instantiates, so the preferred size arguably ought to be respected when the implicit `Window` is synthesised. Recorded here as a measurement rather than a nudge — the "not a high priority" call is yours.
